cmd/oras/repository/ls.go

```diff
@@ -50,7 +55,13 @@ Example - List the repositories under the registry that include values lexically
 			return option.Parse(&opts)
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			opts.hostname = args[0]
+			hostname, namespace, _ := strings.Cut(args[0], "/")
+			opts.hostname = hostname
+			opts.namespace = namespace
+			// namespace is invalid if a tag is provided
+			if strings.Contains(namespace, ":") {
+				return fmt.Errorf("invalid namespace")
+			}
```

Comment:
Can we refactor this part into a function so that we can unit test it?

Reply:
Parsing can be simplified in an easier way.

If `args[0]` contains `/`, we can parse it using [ParseReference()](https://pkg.go.dev/oras.land/oras-go/v2@v2.0.0/registry#ParseReference). Otherwise, it is a hostname.

Reply:
Used `ParseReference`. Need to discuss which paths are invalid.